app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitItemAdapter.java

```diff
@@ -72,15 +75,85 @@ protected void onBindViewHolder(@NonNull HabitViewHolder holder,
                                     @NonNull Habit habit) {
         // set necessary elements of the habit
         holder._habitTitle.setText(habit.getTitle());
+
+        // Fetch habit events for a habit from the database
+        habit.setHabitEvents(new HabitEventList());
+        populateEventList(getListFromFirebase(habit), habit);
+
+        // Create a new progress tracker
+        ProgressTracker progressTracker = new ProgressTracker(habit);
+
+        // Get integer representation of progress
+        int progress = (int) progressTracker.calculateProgressPercentage();
+
+        // Set progress in circular progress and text
+        holder._progressBar.setProgress(progress);
+        String progressText = Integer.toString(progress) + "%";
+        holder._progressText.setText(progressText);
+    }
+
+    /**
+     * Creates a query to get the list of habit events from the database
+     * @param parentHabit (Habit) The habit for which events are being queried
+     * @return The query
+     */
+    @NonNull
+    protected Query getListFromFirebase(Habit parentHabit) {
```

Comment:
Since we are fetching events here, can we rename this to `getEventListFromFirebase` instead?

- [ ] rename method